### DISCUSSION TITLE
Advocate for 2nd neighbour included in graph triangulation

### DISCUSSION THREAD

**Discussion by KristinaUlicna** [alan-turing-institute/grace#166] [Ideas]
I'm generating ground truth graphs from a random graph into which I drew 3 circle motifs. I noticed that although the graph has all objects stored with they respective object identities (code snippet): 

https://github.com/alan-turing-institute/grace/blob/a173bc3fad2d3590c57d45872a19a47d6440b455/grace/simulator/simulate_graph.py#L245-L246

...the GT graph is not constructed correctly because some objects will form edges with the closeby noisy nodes due to the Delaunay triangulation:

Ground Truth Graph                       | Triangulated Graph
:--------------------------------------: | :--------------------------------------:
![image](https://github.com/alan-turing-institute/grace/assets/48791041/9e666b2d-a0a6-42ec-aff2-5352146702b7) | ![image](https://github.com/alan-turing-institute/grace/assets/48791041/dbf14f80-a47d-4862-9912-50904c81fa94) 

I think what would really help out here would be a different type of triangulation:
* hyperlinks to 2-hop neighbourhood,
* forming all edges below certain distance threshold,
* etc.

- Important for #152 
- Mentioned in #167 

Perhaps this could be beneficial for the cross-over object connections. 
@marjanfamili tagging you for further discussions.

**Comment by marjanfamili** [comment]
Are we looking at 3 intersecting circles here ? 

if we add 2-hop neighbours, would this allow for more noise to enter the GT?
We could use higher-order Delauny triangulations?

**Comment by quantumjot** [comment]
You could also add the "flips":
https://en.wikipedia.org/wiki/Delaunay_triangulation#Visual_Delaunay_definition:_Flipping

**Reply by marjanfamili** [reply]
https://www.sciencedirect.com/science/article/pii/S092577210100027X 

A second-order would achieve the flips I think
